newIDE/app/src/AssetStore/CreditsPackages/CreditsPackageStoreContext.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+import * as React from 'react';
+import {
+  listListedCreditsPackages,
+  type CreditsPackageListingData,
+} from '../../Utils/GDevelopServices/Shop';
+
+type CreditsPackageStoreState = {|
+  fetchCreditsPackages: () => void,
+  creditsPackageListingDatas: ?Array<CreditsPackageListingData>,
+  error: ?Error,
+|};
+
+export const CreditsPackageStoreContext = React.createContext<CreditsPackageStoreState>(
+  {
+    fetchCreditsPackages: () => {},
+    creditsPackageListingDatas: null,
+    error: null,
+  }
+);
+
+type CreditsPackageStoreStateProviderProps = {|
+  children: React.Node,
+|};
+
+export const CreditsPackageStoreStateProvider = ({
+  children,
+}: CreditsPackageStoreStateProviderProps) => {
+  const [error, setError] = React.useState<?Error>(null);
+  const [
+    creditsPackageListingDatas,
+    setCreditsPackageListingDatas,
+  ] = React.useState<?Array<CreditsPackageListingData>>(null);
+
+  const isLoading = React.useRef<boolean>(false);
+
+  const fetchCreditsPackages = React.useCallback(
+    () => {
+      // If the credit packages are already loaded, don't load them again.
+      if (isLoading.current || creditsPackageListingDatas) return;
+
+      (async () => {
+        setError(null);
+        isLoading.current = true;
+
+        try {
+          const fetchedCreditsPackageListingDatas = await listListedCreditsPackages();
+
+          console.info(
+            `Loaded ${
+              fetchedCreditsPackageListingDatas
+                ? fetchedCreditsPackageListingDatas.length
+                : 0
+            } credit packages from the store.`
+          );
+
+          setCreditsPackageListingDatas(fetchedCreditsPackageListingDatas);
+        } catch (error) {
+          console.error(
+            `Unable to load the credit packages from the store:`,
+            error
+          );
+          setError(error);
+        }
+
+        isLoading.current = false;
+      })();
+    },
+    [isLoading, creditsPackageListingDatas]
+  );
+
+  React.useEffect(
+    () => {
+      if (isLoading.current) return;
+
+      const timeoutId = setTimeout(() => {
+        console.info('Pre-fetching credit packages from the store...');
+        fetchCreditsPackages();
+      }, 5000);
```

Comment:
I wonder if we should put in the same file all the api calls timing we do at the start of the app so that we can clearly see what comes first and what comes last. This way we can try to not make them at the same time to avoid overloading the device

Reply:
I agree there are a few stores pre-fetching data in the app.
Maybe that's for another PR to make them visible in the same file. I'll update this one to fetch it a bit later, as it's not critical for the user journey